openedx/core/lib/api/permissions.py

```diff
@@ -74,3 +74,11 @@ def has_permission(self, request, view):
             return True
 
         return super(IsUserInUrlOrStaff, self).has_permission(request, view)
+
+
+class IsStaffOrReadOnly(permissions.BasePermission):
+    """
+    Permission that checks to see if the user is staff, permitting only read-only access if they are not.
+    """
+    def has_permission(self, request, view):
+        return request.user.is_staff or request.method in permissions.SAFE_METHODS
```

Comment:
@andy-armstrong and @benmcmorran Unfortunately this is the wrong implementation for course staff. "is_staff" is a global staff boolean largely limited to edX employees. We should really be checking if the user is course staff for PATCH updates.